### PR TITLE
Convert literal comprehensive dicts to dict contructs

### PR DESCRIPTION
**Pull request by Twilight0**
It will help maintain python 2.6 compatibility.

### CODE REVIEW

File: src/streamlink/logger.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def getMessage(self):
         return msg
 
 
-class StreamlinkLogger(logging.getLoggerClass()):
+class StreamlinkLogger(logging.getLoggerClass(), object):
```
Comment (beardypig) [inline src/streamlink/logger.py]: Not of the effect of this, it seems fine - but is it really needed?

File: src/streamlink/plugins/vk.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def follow_vk_redirect(cls, url):
         # If this is a 'videos' catalog URL with an video ID in the GET request, get that instead
         parsed_url = urlparse(url)
         if parsed_url.path.startswith('/videos-'):
-            query = {v[0]: v[1] for v in [q.split('=') for q in parsed_url.query.split('&')] if v[0] == 'z'}
+            query = dict((v[0], v[1] for v in [q.split('=') for q in parsed_url.query.split('&')] if v[0] == 'z'))
```
Comment (beardypig) [inline src/streamlink/plugins/vk.py]: Should be `(v[0], v[1])`.

**Comment by beardypig** [comment]
You'll be fighting a losing battle here, I assume this is for older Kodi support?

File: src/streamlink/logger.py (hunk 1 above)
Comment (Twilight0) [reply src/streamlink/logger.py]: This is required for python 2.6, else I was getting TypeError. Such as this: https://stackoverflow.com/questions/1713038/super-fails-with-error-typeerror-argument-1-must-be-type-not-classobj-when

**Comment by Twilight0** [comment]
@beardypig: Absolutely, this is for Kodi.

File: src/streamlink/logger.py (hunk 1 above)
Comment (beardypig) [reply src/streamlink/logger.py]: Fair enough :)

**Comment by beardypig** [comment]
@Twilight0 I think you need to rebase this branch on to master, looks like some things got a bit muddled up...

**Comment by Twilight0** [comment]
I must have done something wrong in the PR itself. Committed directly to master instead of separate branch, I think that's what messes up, isn't it?

**Comment by bastimeyer** [comment]
You have merged the `origin/master` branch into your own branch instead if rebasing it. It doesn't matter if you've added your new commits to your own `master` branch or to another branch. However, **committing changes to your own `master` branch and then pushing it to your fork and submitting a PR is bad**, because you won't be able to pull and push commits from streamlink's master branch to your fork anymore without having conflicts and submitting another pull request will be even more problematic. I'd suggest you close this PR and properly branch off and submit a new one that only contains your own commits. Try to do this (**don't copy and paste this**, just try to understand what I'm trying to say here):

```bash
# set "origin" remote repo (your fork)
git remote add origin git@github.com:twilight0/streamlink.git
# set "upstream" remote repo (streamlink)
git remote add upstream https://github.com/streamlink/streamlink.git
# get the latest updates
git fetch --all

# fix the master branch mess
git checkout upstream/master
git branch --move master master-old
git branch master

# fix your fork's master branch (close this PR first)
git push --force origin master

# create a new topic branch
git checkout -b fix/python2.6-kodi
# "import" only your own commits from the old master branch
git cherry-pick cabd89563d0240585f799af0da1cfed0da476b9a
git cherry-pick f1500db6e09dc3c5d8a2dc907b1ffc330e028964
# push your new topic branch to your fork and then submit a new pull request
git push origin fix/python2.6-kodi
```

If there are merge conflicts in the future, don't merge anything into your topic branch and instead rebase it to `master` after pulling it from upstream.

**Comment by Twilight0** [comment]
Right, I am closing this and going to create a new one with separate branch.

**Comment by Twilight0** [comment]
New PR: https://github.com/streamlink/streamlink/pull/1792

**Pull request closed without merging**